lib/Backend/NativeCodeGenerator.cpp

```diff
@@ -2802,25 +2810,40 @@ void
 NativeCodeGenerator::ProfilePrint()
 {
     Js::ScriptContextProfiler *codegenProfiler = this->backgroundCodeGenProfiler;
+
     if (Js::Configuration::Global.flags.Verbose)
     {
         //Print individual CodegenProfiler information in verbose mode
         while (codegenProfiler)
         {
-            codegenProfiler->ProfilePrint(Js::Configuration::Global.flags.Profile.GetFirstPhase());
+            if (codegenProfiler->IsInitialized())
+            {
+                codegenProfiler->ProfilePrint(Js::Configuration::Global.flags.Profile.GetFirstPhase());
+            }
             codegenProfiler = codegenProfiler->next;
         }
     }
     else
     {
         //Merge all the codegenProfiler for single snapshot.
-        codegenProfiler = codegenProfiler->next;
-        while (codegenProfiler)
+        Js::ScriptContextProfiler* mergeToProfiler = codegenProfiler;
+        while (mergeToProfiler != nullptr && !mergeToProfiler->IsInitialized())
```

Comment:
Why an extra loop here to find the first initialized profiler?
The loop below should be able to handle that already.  
This loop seems redundant.